### PR TITLE
ffmpeg 4.2.3 + dependencies update

### PR DESCRIPTION
**Pull request by th0ma7**
_Motivation:_  Preparation work for next version of ffmpeg.
_Linked issues:_  None currently.

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### Changes
- Update of `x265` to version 3.3
- Update of `x264` to latest stable snapshot
- Update of Intel `libva` and `libva-utils` to version 2.7.1
- Update of Intel `gmmlib` to 20Q1 version 20.1.1
- Update of Intel Media Driver to 20Q1 version 20.1.1
- Update of Intel Media SDK to 20Q1 version 20.1.1
- Update of `libogg` to version 1.3.4
- Update of `fribidi` to version 1.0.9
- Update of `libbluray` to version 1.2.0
- Update of `libaom` to tag v2.0.0-rc1
- Update of `freetype` to version 2.10.2
- Update of `utils-linux` to version 2.35
- Add experimental `svt-av1` support version 0.8.3 (64-bit arches only)
- Add experimental `svt-hevc` support version 1.4.3 (x86_64 Intel arches only)
- Add `+setuid` to `vainfo` binary for ease of use

_Issues fixed upstream:_
- Intel Media Driver 20Q1: issue https://github.com/intel/media-driver/issues/907 fixed with https://github.com/intel/media-driver/commit/88769b23bc0f6ffd813f9aa946b82456d2bd1d77
- Intel Media SDK 20Q1: https://github.com/Intel-Media-SDK/MediaSDK/issues/2100 fixed with https://github.com/Intel-Media-SDK/MediaSDK/pull/2117

_Locally patched issues:_
- Intel Media Driver 20Q1: https://github.com/intel/media-driver/issues/906

### CODE REVIEW

**Comment by EngMarc** [comment]
which is the intel adam package?

**Comment by hgy59** [comment]
> which is the intel adam package?

do you mean intel __atom__ instead?
then use the (generic) x64 package

**Comment by EngMarc** [comment]
sorry yes intel atom - autocorrect hell

**Comment by hgy59** [comment]
and there is one exception: for `Intel Atom CE5335` (DS-415play and DS-214play) this is `Evansport` not `x64`. 
It is better to reference a Diskstation model or CPU arch than a CPU type.
All models, CPU archs and package archs are listed on the wiki page [Synology and SynoCommunity Package Architectures](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures).

**Comment by th0ma7** [comment]
Thnx @hgy59 
> and there is one exception: for `Intel Atom CE5335` (DS-415play and DS-214play) this is `Evansport` not `x64`.
> It is better to reference a Diskstation model or CPU arch than a CPU type.
> All models, CPU archs and package archs are listed on the wiki page [Synology and SynoCommunity Package Architectures](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures).

File: spk/ffmpeg/src/service-setup.sh [hunk 1]
```diff
@@ -5,10 +5,12 @@ service_postinst ()
...
     chmod u+s /var/packages/ffmpeg/target/bin/ffmpeg
+    chmod u+s /var/packages/ffmpeg/target/bin/vainfo
 }
 
 service_postupgrade ()
 {
     # setuid for proper vaapi access
     chmod u+s /var/packages/ffmpeg/target/bin/ffmpeg
+    chmod u+s /var/packages/ffmpeg/target/bin/vainfo
 }
```
Comment (ymartin59) [inline spk/ffmpeg/src/service-setup.sh:16]: According to https://help.synology.com/developer-guide/synology_package/scripts.html `service_postupgrade` may be discarded.

File: cross/svt-av1/digests [hunk 2]
```diff
@@ -0,0 +1,3 @@
+v0.8.3.tar.gz SHA1 ea792f98e3fca1a3510e73b2c3620ee8a7fd6622
```
[the remark this answers is not in the document]
Comment (th0ma7) [reply cross/svt-av1/digests:1]: i know and hate it as well.  I think I will reach out to the project and issue a bug for this so it gets fixed for next releases.

[the remark this answers is not in the document]
Comment (th0ma7) [reply cross/svt-av1/digests:1]: https://github.com/OpenVisualCloud/SVT-AV1/issues/1319

File: cross/svt-hevc/Makefile [hunk 3]
```diff
@@ -0,0 +1,61 @@
+PKG_NAME = SVT-HEVC
+PKG_VERS = 1.4.3
+PKG_EXT = tar.gz
+PKG_DIST_NAME = v$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://github.com/OpenVisualCloud/SVT-HEVC/archive
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
```
[the remark this answers is not in the document]
Comment (th0ma7) [reply cross/svt-hevc/Makefile:6]: https://github.com/OpenVisualCloud/SVT-HEVC/issues/539

File: cross/svt-hevc/digests [hunk 4]
```diff
@@ -0,0 +1,3 @@
+v1.4.3.tar.gz SHA1 92fdb539cc8d161eef279f12fa888b7a97af95d3
```
[the remark this answers is not in the document]
Comment (th0ma7) [reply cross/svt-hevc/digests:1]: https://github.com/OpenVisualCloud/SVT-HEVC/issues/539

File: cross/svt-av1/digests (hunk 2 above)
[the remark this answers is not in the document]
Comment (th0ma7) [reply cross/svt-av1/digests]: done

File: cross/svt-av1/Makefile [hunk 5]
```diff
@@ -0,0 +1,88 @@
+PKG_NAME = SVT-AV1
+PKG_VERS = 0.8.3
+PKG_EXT = tar.gz
+PKG_DIST_NAME = v$(PKG_VERS).$(PKG_EXT)
```
[the remark this answers is not in the document]
Comment (th0ma7) [reply cross/svt-av1/Makefile:4]: done

File: cross/svt-hevc/digests (hunk 4 above)
[the remark this answers is not in the document]
Comment (th0ma7) [reply cross/svt-hevc/digests]: done

File: cross/svt-hevc/Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (th0ma7) [reply cross/svt-hevc/Makefile:6]: done

File: spk/ffmpeg/src/service-setup.sh (hunk 1 above)
Comment (th0ma7) [reply spk/ffmpeg/src/service-setup.sh:16]: But for now it still stands, right?

Comment (ymartin59) [reply spk/ffmpeg/src/service-setup.sh:16]: Yes, it is harmless but also useless as postinst is called by postupgrade during upgrade sequence.

**Comment by ymartin59** [comment]
@th0ma7 Thanks. Merged. You can go on with CMake support!

**Comment by th0ma7** [comment]
Thnx, next on my todo list!

**Comment by EngMarc** [comment]
Just upgraded to DSM 7.0 so checking for ffmpeg compatibility. I had dl’d the build from the community site but this doesn’t work on DSM 7.0. 
Ideas?
I’m going to do some searching

**Comment by EngMarc** [comment]
Here’s the error basically says ‘Incompatible file format’

Screen Shot 2021-06-13 at 17 42 08

**Comment by hgy59** [comment]
you might find the error cause in the file `/var/log/messages`

**Comment by EngMarc** [comment]
Thanks, I did check the /var/log/messages file. There’s a few errors that videostation erroring out trying generate what looks like thumbnails at first glance. Then, when I do the ffmpeg repair that generates the above error I get this. Not sure it’s related but doing a tail -f on the messages file watching as I try to ‘repair’ the ffmpeg gens the below log messages.

2021-06-14T04:09:07-05:00 DiskStation sscamerad[5957]: SYSTEM:	Last message 'ssdebuglog.cpp:226:S' repeated 10043 times, suppressed by syslog-ng on DiskStation
2021-06-14T04:09:07-05:00 DiskStation sscamerad[7733]: ssdebuglog.cpp:226:SSVPrintf(): open sslog file failed with errno [24].
2021-06-14T04:09:07-05:00 DiskStation sscamerad[7733]: ssdebuglog.cpp:226:SSVPrintf(): open sslog file failed with errno [24].
2021-06-14T04:09:07-05:00 DiskStation sscamerad[7733]: ssdebuglog.cpp:226:SSVPrintf(): open sslog file failed with errno [24].

**Comment by ymartin59** [comment]
Please do not comment on old+merged PR... just create a new issue with reference to PR !

**Pull request merged**